open_spiel/python/games/dynamic_routing/dynamic_routing_game.py

```diff
@@ -0,0 +1,489 @@
+    current volume on its road section and the exit function of the road
+    section (variant of the volume delay function).
+    One vehicle travel time is equal to the time step when they first reach
+    their destination. Therefore the game is simultaneous, explicitly
+    stochastic, is a general sum game with a terminal reward model.
+    See file docstring for more information.
+
+    Attributes inherited from GameInfo:
+        max_chance_outcome: maximum number of chance possibilities. This is
+            equal to 2**num_player as each vehicle can either move or be stuck
+            in traffic.
+        max_game_length: maximum number of time step played. Chosed during by
+            the constructor.
+        max_utility: maximum utility is the opposite of the minimum travel
+            time. Set to 0.
+        min_utility: minimum utility is the opposite of the maximum travel
+            time. Set to - max_game_length - 1.
+        num_distinct_actions: maximum number of action possibles. This is
+            equal to the number of links + 1 (corresponding to having no
+            possible action _NO_POSSIBLE_ACTION).
+        num_player: the number of vehicles. Choosen during by the constructor
+            as the number of vehicles.
+    Attributes:
+        network: the network of the game.
+        vehicles: a list of the vehicle. Their origin and their destination
+            should be road sections of the game. The number of vehicles in the
+            list set the num_player attribute.
+        perform_sanity_checks: if true, sanity checks are done during the game,
+            should be set to false to faster the game.
+    """
+    def __init__(
+        self, params: Mapping[str, Any] = None,
+        network: Network = None,
+        vehicles: List[Vehicle] = None,
+        max_num_time_step: int = 2,
+        perform_sanity_checks: bool = True
+    ):
+        """Initiliaze the game.
+
+        Args:
+            params: game parameters.
+            network: the network of the game.
+            vehicles: a list of the vehicle. Their origin and their destination
+                should be road sections of the game. The number of vehicles in
+                the list set the num_player attribute.
+            max_num_time_step: set the max_game_length attribute.
+            perform_sanity_checks: if true, sanity checks are done during the
+                game, should be set to false to faster the game.
+        """
+        self.network = network if network else _DEFAULT_NETWORK
+        self._vehicles = vehicles if vehicles else _DEFAULT_VEHICLES
+        self.network.check_list_of_vehicles_is_correct(self._vehicles)
+        self.perform_sanity_checks = perform_sanity_checks
+        game_info = pyspiel.GameInfo(
+            num_distinct_actions=self.network.num_links() + INDEX_FIRST_ACTION,
+            max_chance_outcomes=2**len(self._vehicles),
```

Comment:
If this ever becomes a bottleneck, it would be possible to make the game SAMPLED_STOCHASTIC and deal with the chance outcomes in O(num_vehicles) in ApplyAction (e.g. draw from uniform([0,1]) once for each vehicle, without having to build the 2^n possible outcomes in advance)